docs/reference/connectors/snowflake/configuration.mdx

```diff
@@ -0,0 +1,209 @@
+keywords:
+  - snowflake
+  - configuration
+---
+
+# Configuration Reference
+
+## Introduction
+
+The configuration is a metadata object that lists all the database entities — such as tables — that the data connector
+has to know about in order to serve queries. It never changes during the lifetime of the data connector service
+instance. When your database schema changes you will have to update the configuration accordingly, see
+[updating with introspection](#updating-with-introspection).
+
+## Structure
+
+The configuration object is a JSON object with the following fields:
+
+```json
+{
+  "jdbcUrl": "",
+  "jdbcProperties": {},
+  "schemas": [],
+  "tables": [],
+  "functions": [],
+  "nativeQueries": {}
+}
+```
+
+### Property: JDBC URL
+
+The JDBC connection URL to connect to the Snowflake instance. This is a required field.
+
+The value can either be a literal string, or a reference to an environment variable:
+
+```json
+{
+  "jdbcUrl": "jdbc:snowflake://<account-identifier.<region>.snowflakecomputing.com?user=YOUR_USERNAME&&password=YOUR_PASSWORD&db=YOUR_DATABASE&warehouse=YOUR_WAREHOUSE&schema=YOUR_SCHEMA&role=YOUR_ROLE",
+  "jdbcUrl": { "variable": "SNOWFLAKE_JDBC_URL" }
+}
+```
+
+### Property: JDBC Properties
+
+This is a JSON object containing key-value pairs of additional properties to be passed to the JDBC driver.
+
+### Property: Schemas
+
+This is an optional array of schema names to include in the introspection process. If not provided, all schemas will be
+included. **Any schema passed in the JDBC URL will take precedence.**
+
+Example:
+
+```json
+{
+  "schemas": ["PUBLIC", "OTHER_SCHEMA"]
+}
+```
+
+### Property: Tables
+
+This is an array of table definitions, generated automatically during introspection.
+
+Example:
+
+```json
+{
+  "tableName": "DOCS.PUBLIC.USERS",
+  "tableType": "TABLE",
+  "description": null,
+  "columns": [
+    {
+      "name": "AGE",
+      "description": null,
+      "type": "NUMBER",
+      "numeric_precision": 38,
+      "numeric_scale": 0,
+      "nullable": false,
+      "auto_increment": false,
+      "is_primarykey": null
+    },
+    {
+      "name": "NAME",
+      "description": null,
+      "type": "TEXT",
+      "numeric_precision": null,
+      "numeric_scale": null,
+      "nullable": false,
+      "auto_increment": false,
+      "is_primarykey": null
+    },
+    {
+      "name": "ID",
+      "description": null,
+      "type": "NUMBER",
+      "numeric_precision": 38,
+      "numeric_scale": 0,
+      "nullable": false,
+      "auto_increment": true,
+      "is_primarykey": null
+    }
+  ],
+  "pks": [],
+  "fks": null
+}
+```
+
+### Property: Functions
+
+This is an array of function definitions.
+
+Example:
+
+```json
+{
+  "function_catalog": "PUBLIC",
+  "function_schema": "PUBLIC",
+  "function_name": "add",
+  "argument_signature": "(N NUMBER, M NUMBER)",
+  "data_type": "TABLE (N NUMBER, M NUMBER)",
+  "comment": "Adds two numbers"
+}
+```
+
+### Property: Native Queries
```

Comment:
How to work with native operations will be incoming later right? Should we mention if native mutations are supported?